api/server/router/network/filter.go

```diff
@@ -93,18 +48,40 @@ func filterNetworks(nws []libnetwork.Network, filter filters.Args) ([]libnetwork
 	}
 
 	var displayNet []libnetwork.Network
-	for fkey, fhandler := range supportedFilters {
-		errFilter := filter.WalkValues(fkey, func(fval string) error {
-			passList, err := fhandler(nws, fval)
+	for _, nw := range nws {
+		if filter.Include("name") {
+			if !filter.Match("name", nw.Name()) {
+				continue
+			}
+		}
+		if filter.Include("id") {
+			if !filter.Match("id", nw.ID()) {
+				continue
+			}
+		}
+		if filter.Include("label") {
+			if !filter.MatchKVList("label", nw.Info().Labels()) {
+				continue
+			}
+		}
+		displayNet = append(displayNet, nw)
+	}
+
+	if filter.Include("type") {
+		var typeNet []libnetwork.Network
+		errFilter := filter.WalkValues("type", func(fval string) error {
```

Comment:
Is there a reason for using two different approaches now for filtering? i.e. the loop above, and the `filter.WalkValues()` approach here?